go/vt/srvtopo/resilient_server.go

```diff
@@ -28,14 +28,16 @@ import (
 	"golang.org/x/net/context"
 
 	"github.com/youtube/vitess/go/stats"
+	"github.com/youtube/vitess/go/sync2"
 	"github.com/youtube/vitess/go/vt/topo"
 
 	topodatapb "github.com/youtube/vitess/go/vt/proto/topodata"
 	vschemapb "github.com/youtube/vitess/go/vt/proto/vschema"
 )
 
 var (
-	srvTopoCacheTTL = flag.Duration("srv_topo_cache_ttl", 1*time.Second, "how long to use cached entries for topology")
+	srvTopoCacheTTL     = flag.Duration("srv_topo_cache_ttl", 1*time.Second, "how long to use cached entries for topology")
+	srvTopoCacheRefresh = flag.Duration("srv_topo_cache_refresh", 1*time.Second, "how frequently to refresh the topology for cached entries")
```

Comment:
I wonder where to add this, but it would be nice to have scenarios that describe the behavior, something like:

* for entries we don't watch (like the list of Keyspaces), we refresh the list only after srv_topo_cache_ttl.
* for entries we watch, and setting the watch fails, we will use the last know value for srv_topo_cache_ttl.

Reply:
Good idea. I added a block comment describing the behavior in the code itself. I can try to edit the flag description to be more clear but I would worry that would be too long to be useful.